docs/utxo-hd/v0.1/for-developers.md

```diff
@@ -0,0 +1,456 @@
+node. Before UTxO-HD, the values were just in memory already so we didn't need
+to do any further work to bring them in. Now we even have (depending on the
+backing store chosen) to perform I/O to read those.
+
+Here we present a diagram in which the yellow nodes are operations, and the
+`DbChangelog` and `BackingStore` are depicted with colors.
+
+```mermaid
+stateDiagram-v2
+    classDef ledgerdb fill:#556B2F
+    classDef backingstore fill:#8B0000
+    state "block :: blk" as block
+    state "keys :: LedgerTables l KeysMK" as neededEntries
+    block --> getBlockKeySets:::operation
+    dbChangelog:::ledgerdb --> rollback
+    getBlockKeySets --> neededEntries
+    neededEntries --> readKeySets:::operation
+    backingStore:::backingstore --> readKeySets
+    readKeySets --> unforwardedTables
+    state "unforwardedTables :: LedgerTables l ValuesMK" as unforwardedTables
+    block --> rollback:::operation
+    rollback --> truncatedDbChangelog 
+    state "emptyInputState :: l EmptyMK" as emptyInputState
+    truncatedDbChangelog --> current:::operation
+    truncatedDbChangelog --> getDiffs:::operation
+    current --> emptyInputState
+    classDef operation fill:#BDB76B
+    state "seqDiff :: LedgerTables l SeqDiffMK" as seqDiff
+    getDiffs:::dbchangelog --> seqDiff
+    seqDiff --> forwarding:::operation
+    unforwardedTables --> forwarding
+    state "forwardedTables :: LedgerTables l ValuesMK" as forwardedTables
+    forwarding --> forwardedTables
+    emptyInputState --> withLedgerTables:::operation
+    forwardedTables --> withLedgerTables
+    withLedgerTables --> inputState
+    state "inputState :: l ValuesMK" as inputState
+    inputState --> stow:::operation
+    state "stowedState :: l EmptyMK" as stowedState
+    stow --> stowedState
+    block --> apply:::operation
+    stowedState --> apply
+    state "resultingState :: l EmptyMK" as resultingState
+    apply --> resultingState
+    resultingState --> unstow:::operation
+    state "unstowedState :: l ValuesMK" as unstowedState
+    unstow --> unstowedState
+    unstowedState --> projectLedgerTables:::operation
+    state "resultingTables :: LedgerTables l ValuesMK" as resultingTables 
+    projectLedgerTables --> resultingTables
+    unstowedState --> forgetLedgerTables:::operation
+    state "emptyResultingState :: l EmptyMK" as emptyResultingState
+    forgetLedgerTables --> emptyResultingState 
+    state "dbChangelog" as dbChangelog2
+    emptyResultingState --> withLedgerTables2
+    resultingTables --> diff:::operation
+    forwardedTables --> diff
+    state "diffTables :: LedgerTables l DiffMK" as diffTables
+    diff --> diffTables
+    state "withLedgerTables" as withLedgerTables2
+    diffTables --> withLedgerTables2:::operation
+    state "diffState :: l DiffMK" as diffState
+    withLedgerTables2 --> diffState
+    diffState --> dbChangelog2:::ledgerdb: to store
+```
+
+### Era boundaries
```

Comment:
I think this section is redundant, because the next section repeats some of it and supersedes the rest.

Reply:
Indeed it is an artifact of a bad copy-paste on my side